src/main/java/seedu/address/model/FriendlyLink.java

```diff
@@ -227,25 +229,37 @@ public void addPair(Pair pair) {
      * @param elderlyNric Nric of elderly.
      * @param volunteerNric Nric of volunteer.
      */
-    public void addPair(Nric elderlyNric, Nric volunteerNric) {
+
+    public boolean addPair(Nric elderlyNric, Nric volunteerNric) {
         Elderly elderly = getElderly(elderlyNric);
         Volunteer volunteer = getVolunteer(volunteerNric);
 
+        Region elderlyRegion = elderly.getRegion();
+        Region volunteerRegion = volunteer.getRegion();
+
         Set<AvailableDate> elderlyAvailableDates = elderly.getAvailableDates();
         Set<AvailableDate> volunteerAvailableDates = volunteer.getAvailableDates();
 
+        // check if region match. If does not match, issue a warning in feedback
+        boolean issueWarning;
```

Comment:
This method to issue warning from friendlylink seems neat!
Have you considered the way we can issue different types of warnings in the future?

Reply:
I feel we can possibly do that. Currently, "Region does not match" seems the only situation that we issue warning but does not prevent paring. Do you feel there would be possibly other scenarios?